src/GenericContainer/include/BipedalLocomotion/GenericContainer/Vector.h

```diff
@@ -820,5 +873,156 @@ typename Vector<const typename container_data<Class>::type>::eigen_map_const_typ
 
 }
 
+/**
+ * The class ref is used a substitution to a classical reference to a Vector. The advantage of using this,
+ * is that custom vectors (all those supported by GenericContainer::Vector) can be implicitly casted to Ref.
+ * Ref does not allocate any memory in construction, hence can be used as a parameter to be passed by copy.
+ * The = operator clones the content.
+ * Ref inherits Vector<T>, hence it can be used as it was a Vector<T>.
+ */
+template <typename T>
+class BipedalLocomotion::GenericContainer::Vector<T>::Ref : public BipedalLocomotion::GenericContainer::Vector<T>
+{
+public:
+
+    /**
+     * @brief A reference cannot exist on its own.
+     */
+    Ref() = delete;
+
+    /**
+     * @brief Copy constructor
+     * @param other The ref from which to copy the context.
+     */
+    Ref(BipedalLocomotion::GenericContainer::Vector<T>::Ref& other)
+    {
+        m_span = other.m_span;
+        m_resizeLambda = other.m_resizeLambda;
+    }
+
+    /**
+     * @brief Move constructor
+     * @param other The ref from which to get the context.
+     */
+    Ref(BipedalLocomotion::GenericContainer::Vector<T>::Ref&& other)
+    {
+        m_span = other.m_span;
+        m_resizeLambda = other.m_resizeLambda;
+    }
+
+    /**
+     * @brief Constructor from a GenericContainer::Vector<T>&
+     * @param other The input vector from which the context is copied
+     */
+    Ref(BipedalLocomotion::GenericContainer::Vector<T>& other)
+    {
+        m_span = other.m_span;
+        m_resizeLambda = other.m_resizeLambda;
+    }
+
+    /**
+     * @brief Constructor from a GenericContainer::Vector<T>&
+     * @param other The input vector from which the context is taken
+     * In principle, Ref should be the reference of a Vector<T> which should remain alive while
+     * Ref is alive. On the other hand, Vector<T> is only a pointer to some data which does not own.
+     * Hence, Ref can remain alive even if the Vector<T> is deleted, provided that the original container
+     * stays alive.
+     */
+    Ref(BipedalLocomotion::GenericContainer::Vector<T>&& other)
+    {
+        m_span = other.m_span;
+        m_resizeLambda = other.m_resizeLambda;
+    }
+
+    /**
+     * Constructor from another container.
+     * This is used if:
+     * - the input container is not a GenericContainer::Vector, to avoid ambiguities with other constructors
+     * - the input container is not a string. This allows using Ref and string with overloaded methods.
+     * - a GenericContainer::Vector<T> can be constructed from the Container
+     * - T is not const
+     * - the input container is not const.
+     */
+    template <class Vector, typename = typename std::enable_if<!GenericContainer::is_vector<Vector>::value &&
+                                                               !std::is_same<Vector, std::string>::value &&
```

Comment:
in C++17 we have `is_same_v` which can replace `is_same<T, T>::value`

Reply:
Yes, it is a little more verbose in this way, but some static analyzers complain less sometimes :grin: